### DISCUSSION TITLE
LiveDanmaku 的 DISCONNECT 事件没有触发

### DISCUSSION THREAD

**Discussion by tp1415926535** [Nemo2011/bilibili-api#763] [Q&A] [answered]
根据[文档](https://nemo2011.github.io/bilibili-api/#/modules/live?id=class-livedanmaku)有个DISCONNECT事件可以监听：
``` python
@danmaku.on('DISCONNECT')
async def on_disconnect(event):
    print(event)
```
但是我断开连接后只有两条框架自带的日志：
```
[22363255][2024-05-18 21:16:18,582][INFO] 连接正在关闭
[22363255][2024-05-18 21:16:18,582][INFO] 连接已关闭
```
没有触发DISCONNECT里的事件，需要怎么做才能触发

**Comment by z0z0r4** [comment, marked as answer]
此处的事件都是B站传过来的CMD吧，你主动断开连接，又不是B站断的

**Reply by tp1415926535** [reply]
这样吗… 懂了，那应该在`loop.run_until_complete(danmaku.connect())`后面等它结束，而不是`loop.create_task(danmaku.disconnect())`后面或者`@danmaku.on('DISCONNECT')`事件里面
